resources/templates/cloudformation/auto_scaling_group.yaml

```diff
@@ -1,87 +1,21 @@
-  HealthCheckGracePeriod:
-    Description: Length of time in seconds after which health checks start
-    Type: Number
-  IAMInstanceProfileName:
-    Description: IAM profile name for the instances
-    Type: String
-  ImageID:
-    Description: CoreOS AMI for the instances
-    Type: String
-  InstanceType:
-    Description: EC2 instance type
-    Type: String
-  KeyName:
-    Description: The name of the KeyPair containing the SSH key
-    Type: String
-  LoadBalancerName:
-    Description: The name of the load balancer for the instances
-    Type: String
-  SecurityGroupID:
-    Description: The ID of the security group for the instances
-    Type: String
-  SmallCloudConfig:
-    Description: The userdata provided to the instances
-    Type: String
-  SubnetID:
-    Description: The ID of the Subnet in the VPC
-    Type: "AWS::EC2::Subnet::Id"
-  VPCID:
-    Description: The ID of the VPC
-    Type: String
-Resources:
-  {{ .ASGType }}LaunchConfiguration:
-    Type: "AWS::AutoScaling::LaunchConfiguration"
-    Description: {{ .ASGType }} launch configuration
-    Properties:
-      ImageId: !Ref ImageID
-      SecurityGroups:
-      - !Ref SecurityGroupID
-      InstanceType: !Ref InstanceType
-      IamInstanceProfile: !Ref IAMInstanceProfileName
-      BlockDeviceMappings:
-      {{ range .BlockDeviceMappings }}
-      - DeviceName: "{{ .DeviceName }}"
-        Ebs:
-          DeleteOnTermination: {{ .DeleteOnTermination }}
-          VolumeSize: {{ .VolumeSize }}
-          VolumeType: {{ .VolumeType }}
-      {{ end }}
-      AssociatePublicIpAddress: !Ref AssociatePublicIPAddress
-      UserData: !Ref SmallCloudConfig
-      KeyName: !Ref KeyName
-  {{ .ASGType }}AutoScalingGroup:
+{{define "autoscaling_group"}}{{ .ASGType }}AutoScalingGroup:
     Type: "AWS::AutoScaling::AutoScalingGroup"
     Properties:
       VPCZoneIdentifier:
-        - !Ref SubnetID
-      AvailabilityZones:
-        - !Ref AZ
-      MinSize: !Ref ASGMinSize
-      MaxSize: !Ref ASGMaxSize
+        - {{ .SubnetID }}
+      AvailabilityZones: [{{ .AZ }}]
+      MinSize: {{ .ASGMinSize }}
+      MaxSize: {{ .ASGMaxSize }}
       LaunchConfigurationName: !Ref {{ .ASGType }}LaunchConfiguration
       LoadBalancerNames:
-        - !Ref LoadBalancerName
-      HealthCheckGracePeriod: !Ref HealthCheckGracePeriod
+        - {{ .LoadBalancerName }}
+      HealthCheckGracePeriod: {{ .HealthCheckGracePeriod }}
     UpdatePolicy:
       AutoScalingRollingUpdate:
         # minimum amount of instances that must always be running during a rolling update
-        MinInstancesInService: 2
+        MinInstancesInService: {{ .MinInstancesInService }}
         # only do a rolling update of this amount of instances max
-        MaxBatchSize: 2
+        MaxBatchSize: {{ .MaxBatchSize }}
         # after creating a new instance, pause operations on the ASG for this amount of time
-        PauseTime: PT10S
+        PauseTime: {{ .RollingUpdatePauseTime }}
+{{end}}
```

Comment:
I like the new template files but could you change the path to `service/templates/cloudformation`? We want to remove the resources dir and have everything under `service/` to match the other operators.

Also later we'll need to support multiple versions of the templates. Once we've implemented versioning the Cloud Formation resource in operatorkit we could use the same version in the templates dir. WDYT?

Reply:
Sure, makes perfect sense. I've also modified the Dockerfile and tested it, worked but putting the resources out of `/opt` so that the relative path works both in the container and in unit test runs, pls take look.